### ISSUE TITLE
systemctl status: non-existing slice or device units are shown as loaded

### ISSUE THREAD

**Issue by yuwata** [systemd/systemd#7370]
### Submission type

 - Bug report

### systemd version the issue has been seen with

git snapshot 83123ab with some PRs

### Used distribution

Fedora 27 x86_64

### In case of bug report: Expected behaviour you didn't see

```
$ systemctl status foo.automount
Unit foo.automount could not be found.
$ systemctl status foo.slice
Unit foo.slice could not be found.
```

### In case of bug report: Unexpected behaviour you saw

```
$ systemctl status foo.automount
● hoge.automount
   Loaded: loaded
   Active: inactive (dead)
    Where: /foo
$ systemctl status foo.slice
● foo.slice
   Loaded: loaded
   Active: inactive (dead)

```
Note that the output of `systemd-analyze dump` does not contain the units.
Are there any reason that automount and slice units are always shown as loaded? Or is it a bug?

**Comment by yuwata** [comment]
If the following command is run while home.automount does not exist,
`busctl introspect org.freedesktop.systemd1 /org/freedesktop/systemd1/unit/home_2eautomount`
then pid1 adds additional dependencies to other units, especially to `home.mount`.
Is it desired?

If I understand correctly, an arbitrary automount unit has a corresponding unit file. While mount unit file may not exist as some units are generated from /proc/self/mountinfo. Right?

**Comment by yuwata** [comment]
The following units unconditionally call `unit_load_fragment_and_dropin_optional()`:
- [x] automount
- [ ] device
- [ ] slice
- [x] swap

The checked unit types are fixed by #7387.

**Comment by marcosfrm** [comment]
Related to https://github.com/systemd/systemd/issues/6956 ?

**Comment by yuwata** [comment]
@marcosfrm I am not sure what causes #6956, but I guess it is not related to this. Because 1) if we do not run `systemctl home.automount` then no new unwanted dependency is created, and 2) even if such a dependency is created, this does not disturb the shutdown process, at least my environment.

BTW, #7387 just fixes part of this. Still slice and device units are shown to be loaded even if the corresponding unit files, cgroup, or devices do not exist. So let us reopen this.

**Comment by yuwata** [comment]
Loading slice and device units without preparation does not create any new dependencies, thus it seems harmless. Now let's close this.

**Comment by r7vme** [comment]
Hello, we experienced consequences of this bug for `device` units.

We have Kubernetes cluster. Calico creates and deletes a lot of tap interfaces. Problem is that after a while we got 4k orphaned units in systemd and this cause metrics collector timeouts (that's why we found this issue.)

Workaround was to reload systemd. We would prefer this bug to be fixed in `systemd` instead of creating own "dirty" hacks to reload systemd

logs:
```
joe@ip-10-1-3-103 ~ $ systemctl list-units --no-pager | grep sys-devices-virtual-net-cali | wc -l
4094
```
Most of the tap devices are not exist already.
```
joe@ip-10-1-3-103 ~ $ sudo ls /sys/devices/virtual/net/calif1abab2cc93
ls: cannot access '/sys/devices/virtual/net/calif1abab2cc93': No such file or directory
```
